src/fhir/api.ts

```diff
@@ -136,7 +136,23 @@ export const checkResponseOK = async (response: Response) => {
     console.error(r.body);
     throw new Error(response.statusText);
   }
-  if (!(r.type === "bundle-response")) {
+  if (r.type === "batch-response") {
+    const errorsForModal: string[] = [];
+    const errorResponses = (r as BundleResponse).entry
+      .map((entry) => entry.response)
+      .filter((response) => !response.status.toString().startsWith("2"));
+    if (errorResponses.length > 1) {
+      errorResponses.forEach((error) => {
+        const errorDetails: any = [
+          r.resourceType.toString(),
+          error.status.toString(),
+          error.outcome.issue?.[0].diagnostics.toString(),
+        ];
+        errorsForModal.push(errorDetails); //had as string[] but errorDetails claimed it couldn't be assigned to type string
```

Comment:
makes sense because we're adding an array of strings to an array (so the errors for modal would have to be an array of an array of strings). Better to make a custom type which has the resource, error status, and diagnostic. Then have the errorsForModal as an array of that custom type